truffle/src/com.oracle.truffle.dsl.processor/src/com/oracle/truffle/dsl/processor/model/SpecializationData.java

```diff
@@ -353,10 +355,18 @@ public void setReachable(boolean reachable) {
         this.reachable = reachable;
     }
 
+    public void setUncached(boolean uncached) {
```

Comment:
Maybe rename this field reachableUncached to makle it clearer what it means.
A specialization is not uncached by itself.
At the same time it would make sense to rename SpecializationData#reachable to reachableCached.

Reply:
This would be confusing. The `isUncached` field does not mean "is it reachable in the uncached node?", but rather "is this specialization taken into account when generating the uncached node?". It has nothing to do with actual reachability, just a helper field.
I've only cached it here to avoid some map lookups, but honestly maybe we'd better off just doing those lookups and not add more fields here?